fix null BYTES value insert/upsert for SQL Server

nwt opened the pull request:
## Problem

A sink that sends a null `BYTES` value to SQL Server will throw a `java.sql.BatchUpdateException` with reason "Implicit conversion from data type nvarchar to varbinary(max) is not allowed. Use the CONVERT function to run this query."

This is the problem described in issue #1138.

## Solution

Provide an explicit SQL type for null `BYTES` values in `SqlServerDatabaseDialect.getSqlTypeForSchema()`.

##### Does this solution apply anywhere else?
- [ ] yes
- [x] no

## Test Strategy

##### Testing done:
- [ ] Unit tests
- [x] Integration tests
- [ ] System tests
- [ ] Manual tests

## Release Plan
?

nwt commented:
Hello?

nwt commented:
Is there anything I can do to move this along?

sgv0007 commented:
@nwt  I am using kafka-connect-JDBC-10.4.1 and i still see this issue . in latest code under SQLServerDialect , i dont see the code fix you added .

 @Override
  protected Integer getSqlTypeForSchema(Schema schema) {
    return schema.type() == Schema.Type.BYTES ? Types.VARBINARY : null;
  }

nwt commented:
@sgv0007: This PR wasn't merged. I closed it because I couldn't get a response from the maintainers here or in Confluent's Slack space.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/confluentinc/kafka-connect-jdbc?pullRequest=1151) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/confluentinc/kafka-connect-jdbc?pullRequest=1151) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/confluentinc/kafka-connect-jdbc?pullRequest=1151) it.</sub>

sagarrao12 reviewed (approved):
LGTM @mukkachaitanya / @rajdangwal you want to make a second pass at this?

sagarrao12 commented:
@nwt , thanks for the PR and your patience and sorry for the delay in reviewing this.

rajdangwal reviewed (approved):
Thanks @nwt 
LGTM.

rajdangwal commented:
@nwt Since this is a bugfix, the PR should target the branch 10.0.x and not master. We will be doing a patch release from there.
cc: @mukkachaitanya

nwt commented:
@sagarrao12 and @rajdangwal: Thank you both for your attention to this!

@rajdangwal: As requested, I've changed the target branch to 10.0.x. (I've also rebased my changes onto that branch.)

sgv0007 commented:
@nwt This issue is not resolved . I am using version 10.7.3 and trying to insert null value into a binary field in destination table and I get below error.

Exception chain:\njava.sql.BatchUpdateException: Implicit conversion from data type nvarchar to binary is not allowed.
